pennylane/templates/subroutines/qsvt.py

```diff
@@ -569,8 +571,154 @@ def _compute_qsp_angle(poly_coeffs):
             )
 
     return rotation_angles
+###########################################################################################################################
+"""
+Implementation of the QSP (Quantum Signal Processing) algorithm proposed in https://arxiv.org/pdf/2002.11649 
+"""
 
+import scipy
 
+def jit_if_available(func):
+    """
+    decorator that jax.jit the qsp_iterative optimization functions if jax is available
+    the static arguments to the jit function are meant to be contained in kwargs
+    """
+    try:
+        from functools import partial
+        from functools import wraps
+        import jax
+
+        jax.config.update("jax_enable_x64", True)
+        return jax.jit(func)
+    except ImportError:
+        return func
+
+
+
+@jit_if_available
+def cheby_pol(x, degree):
```

Comment:
Please see how the doc strings of other functions are doc strings to follow a similar style (“Args”, “Return”,...) 😄

Reply:
This applies also to the other functions